ipykernel/mplshim.py

```diff
@@ -0,0 +1,89 @@
+"""Import hooks to activate our inline backend when matplotlib is loaded
+"""
+
+import imp
+import sys
+
+def set_inline_backend(mpl):
+    from IPython.core.pylabtools import backends, configure_inline_support
+    from .kernelapp import IPKernelApp
+    mpl.interactive(True)
+    mpl.use(backends['inline'])
+    configure_inline_support(IPKernelApp.instance().shell, backends['inline'])
+
+class Py2MPLFinderShim(object):
+    @classmethod
+    def find_module(cls, fullname, path=None):
+        if (fullname != 'matplotlib') or (path is not None):
```

Comment:
Setting the hook on `matplotlib` makes it impossible for 

    import matplotlib
    matplotlib.use('something')

to work. Is there something else we can look for that wouldn't preempt explicit backend selection? I've used `pyplot` in my own version of this, but I'm not sure if it's sufficiently general.